spec/models/my_academics/semesters_spec.rb

```diff
@@ -1,782 +1,664 @@
 describe MyAcademics::Semesters do
```

Comment:
I just want to note that I experienced difficulties fixing the spec for this model after making a minor update to the interface for the data. I decided to just completely re-write the spec for this model. It's still a work in progress that is going to take a while. There are many methods that aren't covered now. Either we can consider this okay, or I can agree to contribute more time for coverage during TST/QAT time (if possible)